metrics/impl/default_compass.go

```diff
@@ -78,6 +78,34 @@ type DefaultCompass struct {
 	reservoir metrics.Reservoir
 }
 
+func (cp *DefaultCompass) GetInstantCountInterval() time.Duration {
+	return cp.bucketInterval
+}
+
+func (cp *DefaultCompass) GetFifteenMinuteRate() float64 {
+	cp.tickIfNecessary()
+	return cp.m15Rate.GetRate(time.Second)
+}
+
+func (cp *DefaultCompass) GetFiveMinuteRate() float64 {
+	cp.tickIfNecessary()
+	return cp.m5Rate.GetRate(time.Second)
+}
+
+func (cp *DefaultCompass) GetMeanRate() float64 {
+	count := cp.GetCount()
+	if count == 0 {
+		return 0
+	}
+	elapsed := cp.clock.GetTick() - cp.startTime
+	return float64(count) / float64(elapsed) * float64(time.Second.Nanoseconds())
```

Comment:
variable 'elapsed'  may be 0?

Reply:
It could be zero or negative. I've fixed it.